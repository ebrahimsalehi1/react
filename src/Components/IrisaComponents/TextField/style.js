import {createStyles} from '@material-ui/core/styles'

export const useStyles = createStyles(theme=>({
    marginFullwidth: {
        margin: '12px 16px 8px',
    },
    margin2: {
        margin: '12px 16px 8px',
        width: '85%'
    },     
  }))
