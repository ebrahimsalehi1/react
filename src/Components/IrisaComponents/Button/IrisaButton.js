import React from 'react'
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'
import {withStyles} from '@material-ui/core'
import {useStyles} from './style'

function IrisaButton(props){

    const {classes,variant,label} = props

    return (<Button
            //className={classes}
            className={"irisabutton"}
            variant={variant}
            data-testid={"IRISA_BUTTON"}
        >{label}</Button>
    )
}

IrisaButton.propTypes = {
    classes: PropTypes.object.isRequired,
};

export default withStyles(useStyles)(IrisaButton);
