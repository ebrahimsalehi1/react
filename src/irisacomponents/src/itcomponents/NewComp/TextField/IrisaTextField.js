
import React,{useState,useEffect} from 'react'
import {TextField,withStyles} from '@material-ui/core'
import {createStyles} from '@material-ui/core/styles'
import validation from '../Validation'
import NumberFormat from 'react-number-format'
import PropTypes from 'prop-types'
import MaskedInput from 'react-text-mask'
import {useStyles} from './style'

IrisaTextField.defaultProps = {
  validationType: [],
  mask: null,
  format:null,
  type:'general',
  variant:'outlined'
  //rules: null,
  //onChange: null,
  //onBlur: null,
}


function IrisaTextField(props){

    //const {rules,classes,fullWidth,onCustomValidation,onChange,onBlur} = props

    const {
      classes, useLov, label, name, disabled, required,
      validationType, validationTypeParam, adornment, icon, 
      select, InputProps, SelectProps,
      fullWidth, rows, value, multiline, variant,className
    } = props;

    const { rules, mask, inputRef, placeholderChar, onCustomValidation, 
      format,prefix,type,
      onChange, onBlur } =props

    let grules = rules   
    if(grules===undefined && validationType!==undefined )  {
      grules ={      
        trigger: 'blur' ,
        validationType:validationType,
        validationParams:validationTypeParam,
        onCustomValidation:true      
      }      
    }

    let resultValidation = undefined
    if(grules!==undefined){
      resultValidation = validation(props.value,grules.validationType,grules.validationParams)
    }

    const [allValidation,setAllValidation] = useState(grules===undefined ? true:resultValidation.isValid)

    function onChangeHandler(event){
        if(grules.trigger==='change' && onChange!==undefined )
          onChange(event);

        if(grules.trigger==='blur' && onBlur!==undefined )
          onBlur(event);

        const {name,value} =event.target
        if(grules!==undefined){
          resultValidation = validation(value,grules.validationType,grules.validationParams)
          setAllValidation(resultValidation.isValid && grules.onCustomValidation)
        }

        //console.log('on changed',resultValidation)        
    }      

    useEffect(()=>{
        console.log('refresh occured ....')
    },allValidation)

    return (<>
    { type==='general' && 
        <TextField 
        variant={variant} 
        //className={classes.textField}
        className={fullWidth ? classes.marginFullwidth : classes.margin2}
        fullWidth={fullWidth}        
        name={name} 
        value={value}
           //className={fullWidth ? classes.marginFullwidth : classes.margin2}
        onChange={grules!==undefined && grules.trigger==='change' ?  onChangeHandler:null } 
        onBlur={grules!==undefined && grules.trigger==='blur' ? onChangeHandler:null}
        error={!allValidation} 
        helperText={!allValidation && resultValidation.messages}
        />
    }
    {
      type==='mask' &&
       <MaskedInput
       variant={variant} 
       className={fullWidth ? classes.marginFullwidth : classes.margin2}
       fullWidth={fullWidth}        
       name={name} 
       value={value}
       ref={inputRef}
       mask={mask}
       placeholderChar={placeholderChar}
       showMask
       onChange={grules!==undefined && grules.trigger==='change' ?  onChangeHandler:null } 
       onBlur={grules!==undefined && grules.trigger==='blur' ? onChangeHandler:null}
     />   
    }
    {
      type==='number' &&
      <NumberFormat
      variant={variant} 
      className={fullWidth ? classes.marginFullwidth : classes.margin2}
      fullWidth={fullWidth}        
      name={name} 
      value={value}
      getInputRef={inputRef}
      format={format}
      thousandSeparator
      prefix={prefix}   
      onValueChange={values => {
        console.log(values.target)
      }}         
      //onChange={rules!==undefined && rules.trigger==='change' ?  onChangeHandler:null } 
      //onBlur={rules!==undefined && rules.trigger==='blur' ? onChangeHandler:null}
      />
    }
    </>
    )
}

IrisaTextField.propTypes = {
  rules: PropTypes.object,
  onChange: PropTypes.func,
  onBlur: PropTypes.func,
  type: PropTypes.oneOf(['general','mask','number']),

  classes: PropTypes.object.isRequired,
  validationType: PropTypes.oneOf(['afterDate', 'alpha', 'alphaFa',
      'alphanumeric', 'alphanumericFa', 'email', 'equals', 'lowercase',
      'matches', 'mobilePhone', 'number', 'numberFloat', 'numberInt', 'required', 'uppercase']),
  validationTypeParam: PropTypes.array,
  label: PropTypes.string,
  name: PropTypes.string,
  adornment: PropTypes.bool,
  icon: PropTypes.object,
  value: PropTypes.string,
  onChange: PropTypes.func,
  fullWidth: PropTypes.bool,
  rows: PropTypes.string,
  multiline: PropTypes.bool,
  useLov: PropTypes.bool

}

export default withStyles(useStyles)(IrisaTextField)
