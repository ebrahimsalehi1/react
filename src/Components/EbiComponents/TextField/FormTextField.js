import React, { useState } from 'react'
import {
// <<<<<<< HEAD
//     Button,
//     Select,
//     FormControl,
//     FormControlLabel,InputLabel,
//     MenuItem,Paper,Checkbox,MuiThemeProvider
// =======
  Button,
  TextField,
  Select,
  FormControl,
  FormControlLabel,
  InputLabel,
  InputBase,
  MenuItem,
  Paper,
  Checkbox,
  MuiThemeProvider,
  Grid,
  createMuiTheme
// >>>>>>> a25a41d8ba08d9297b3576502967b701e2c889de
} from '@material-ui/core'
import IrisaTextField from '../TextField/IrisaTextField'
//import {theme} from '../TextField/style'

//import { CheckBox } from '@material-ui/icons';
import purple from '@material-ui/core/colors/purple';
import green from '@material-ui/core/colors/green';

const styles2 = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
  },
  margin: {
    margin: theme.spacing.unit,
  },
  cssLabel: {
    '&$cssFocused': {
      color: purple[500],
    },
  },
  cssFocused: {},
  cssUnderline: {
    '&:after': {
      borderBottomColor: purple[500],
    },
  },
  cssOutlinedInput: {
    '&$cssFocused $notchedOutline': {
      borderColor: purple[500],
    },
  },
  notchedOutline: {},
  bootstrapRoot: {
    'label + &': {
      marginTop: theme.spacing.unit * 3,
    },
  },
  bootstrapInput: {
    borderRadius: 4,
    position: 'relative',
    backgroundColor: theme.palette.common.white,
    border: '1px solid #ced4da',
    fontSize: 16,
    width: 'auto',
    padding: '10px 12px',
    transition: theme.transitions.create(['border-color', 'box-shadow']),
    // Use the system font instead of the default Roboto font.
    fontFamily: [
      '-apple-system',
      'BlinkMacSystemFont',
      '"Segoe UI"',
      'Roboto',
      '"Helvetica Neue"',
      'Arial',
      'sans-serif',
      '"Apple Color Emoji"',
      '"Segoe UI Emoji"',
      '"Segoe UI Symbol"',
    ].join(','),
    '&:focus': {
      borderRadius: 4,
      borderColor: '#80bdff',
      boxShadow: '0 0 0 0.2rem rgba(0,123,255,.25)',
    },
  },
  bootstrapFormLabel: {
    fontSize: 18,
  },
});


export default function Form1(props) {
  const [state, setState] = useState({
    select: 'standard'
  })

  const classes = styles2(createMuiTheme({
    spacing: {unit:8},
    palette: {
      primary: green,
    },
    typography: { useNextVariants: true },    
  }))

  // setValarray({...valarray,[name]:{
  //     ...valarray[name],value:value
  //   }})

  function handlerStateChange(e) {
    const { name, value } = e.target
    setState({ [name]: value })
    console.log(name, value, state)
  }

  const [valarray, setValarray] = useState({
    txt1: { value: 'text 1' },
    txt2: { value: 'text 2' },
    txt3: { value: 'text 3' },
    txt4: { value: '' },
    txt6: { value: 'text 6' },
    //'RadioValue':'IrisaTextField',
    IrisaDialog: { open: false }
    //'IrisaSelect':{value:''}
  })

  function handlerTextField(event) {
    const { name, value } = event.target
    setValarray({
      ...valarray,
      [name]: {
        ...valarray[name],
        value: value
      }
    })
  }

  const TestTextField = () => (
    <>
      <IrisaTextField
        name={'txt1'}
        value={valarray.txt1.value}
        variant={state.select}
        fullWidth
        rules={{
          trigger: 'change',
          validationType: ['alpha'],
          validationParams: [],
          onCustomValidation: true
        }}
        onChange={handlerTextField}
      />
      <br />
      <IrisaTextField
        name={'txt2'}
        value={valarray.txt2.value}
        variant={state.select}
        rules={{
          trigger: 'blur',
          validationType: ['equals'],
          validationParams: [100],
          onCustomValidation: true
        }}
      />
      <br />
      <IrisaTextField
        name={'txt3'}
        value={valarray.txt3.value}
        variant={state.select}
        rules={{
          trigger: 'blur',
          validationType: ['lowercase'],
          validationParams: [100],
          onCustomValidation: true
        }}
      />

      <br />
      <IrisaTextField
        name={'txt4'}
        value={valarray.txt4.value}
        variant={state.select}
        type={'mask'}
        mask={[
          '(',
          /[1-9]/,
          /\d/,
          /\d/,
          ')',
          ' ',
          /\d/,
          /\d/,
          /\d/,
          '-',
          /\d/,
          /\d/,
          /\d/,
          /\d/
        ]}
      />
      <br />
      <IrisaTextField
        name={'txt5'}
        value={valarray.txt4.value}
        variant={state.select}
        type={'number'}
        prefix={'ریال'}
      />
      <br />
      <IrisaTextField
        name={'txt6'}
        value={valarray.txt4.value}
        variant={state.select}
      />
      <br />
    </>
  )

  const NewDesign = () => (
    <div>
      <Grid container >
        <Grid item xs={6}>
          <IrisaTextField 
          label={'نام'} 
          placeholder={'لطفا نام خود را وارد کنید'}  
          variant={'standard'}/>      
        </Grid>
        <Grid item xs={6}>
          <TextField  label={'*نام'} variant={'standard'} placeholder={'اطفا نام خود را وارد کنید'}  />      
        </Grid>

        <Grid item xs={6}>
          <FormControl className={classes.margin}>
          <InputLabel shrink htmlFor="bootstrap-input" className={classes.bootstrapFormLabel}>
            Bootstrap
          </InputLabel>
          <InputBase
            id="bootstrap-input"
            defaultValue="react-bootstrap"
            classes={{
              root: classes.bootstrapRoot,
              input: classes.bootstrapInput,
            }}
          />
      </FormControl> 
        </Grid>
        <Grid item xs={6}>
          <IrisaTextField  label={'نام'}/>      
        </Grid>

        <Grid item xs={6}>
          <IrisaTextField  label={'نام'}/>      
        </Grid>
        <Grid item xs={6}>
          <IrisaTextField  label={'نام'}/>      
        </Grid>

        <Grid item xs={6}>
          <IrisaTextField  label={'نام'}/>      
        </Grid>
        <Grid item xs={6}>
          <IrisaTextField  label={'نام'}/>      
        </Grid>

      </Grid>
    </div>
  )

  return (
    <div>
      <Paper style={{ width: 500, height: 500 }}>
        {/* <FormControl>
          <InputLabel>Variant</InputLabel>
          <Select
            label={'label1'}
            name={'select'}
            value={state.select}
            onChange={handlerStateChange}>
            <MenuItem value={'standard'} label={'standard'}>
              standard
            </MenuItem>
            <MenuItem value={'outlined'} label={'outlined'}>
              outlined
            </MenuItem>
            <MenuItem value={'filled'} label={'filled'}>
              filled
            </MenuItem>
          </Select>
          <InputLabel>error</InputLabel>
          { <Checkbox label={"error"}/> }
          <br />
          <hr />
          Components Test
          <div>
            <TestTextField />
          </div>
        </FormControl> */}
        <NewDesign />
      </Paper>
    </div>
  )
}
