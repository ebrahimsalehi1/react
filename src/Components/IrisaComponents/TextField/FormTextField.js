import React, { useState } from 'react'
import {
<<<<<<< HEAD
    Button,
    Select,
    FormControl,
    FormControlLabel,InputLabel,
    MenuItem,Paper,Checkbox,MuiThemeProvider
=======
  Button,
  Select,
  FormControl,
  FormControlLabel,
  InputLabel,
  MenuItem,
  Paper,
  Checkbox,
  MuiThemeProvider
>>>>>>> a25a41d8ba08d9297b3576502967b701e2c889de
} from '@material-ui/core'
import IrisaTextField from '../TextField/IrisaTextField'
//import {theme} from '../TextField/style'

//import { CheckBox } from '@material-ui/icons';

export default function Form1(props) {
  const [state, setState] = useState({
    select: 'standard'
  })
  // setValarray({...valarray,[name]:{
  //     ...valarray[name],value:value
  //   }})

  function handlerStateChange(e) {
    const { name, value } = e.target
    setState({ [name]: value })
    console.log(name, value, state)
  }

  const [valarray, setValarray] = useState({
    txt1: { value: 'text 1' },
    txt2: { value: 'text 2' },
    txt3: { value: 'text 3' },
    txt4: { value: '' },
    txt6: { value: 'text 6' },
    //'RadioValue':'IrisaTextField',
    IrisaDialog: { open: false }
    //'IrisaSelect':{value:''}
  })

  function handlerTextField(event) {
    const { name, value } = event.target
    setValarray({
      ...valarray,
      [name]: {
        ...valarray[name],
        value: value
      }
    })
  }

  const TestTextField = () => (
    <>
      <IrisaTextField
        name={'txt1'}
        value={valarray.txt1.value}
        variant={state.select}
        fullWidth
        rules={{
          trigger: 'change',
          validationType: ['alpha'],
          validationParams: [],
          onCustomValidation: true
        }}
        onChange={handlerTextField}
      />
      <br />
      <IrisaTextField
        name={'txt2'}
        value={valarray.txt2.value}
        variant={state.select}
        rules={{
          trigger: 'blur',
          validationType: ['equals'],
          validationParams: [100],
          onCustomValidation: true
        }}
      />
      <br />
      <IrisaTextField
        name={'txt3'}
        value={valarray.txt3.value}
        variant={state.select}
        rules={{
          trigger: 'blur',
          validationType: ['lowercase'],
          validationParams: [100],
          onCustomValidation: true
        }}
      />

      <br />
      <IrisaTextField
        name={'txt4'}
        value={valarray.txt4.value}
        variant={state.select}
        type={'mask'}
        mask={[
          '(',
          /[1-9]/,
          /\d/,
          /\d/,
          ')',
          ' ',
          /\d/,
          /\d/,
          /\d/,
          '-',
          /\d/,
          /\d/,
          /\d/,
          /\d/
        ]}
      />
      <br />
      <IrisaTextField
        name={'txt5'}
        value={valarray.txt4.value}
        variant={state.select}
        type={'number'}
        prefix={'ریال'}
      />
      <br />
    </>
  )

  return (
    <div>
      <Paper style={{ width: 500, height: 500 }}>
        <FormControl>
          <InputLabel>Variant</InputLabel>
          <Select
            label={'label1'}
            name={'select'}
            value={state.select}
            onChange={handlerStateChange}>
            <MenuItem value={'standard'} label={'standard'}>
              standard
            </MenuItem>
            <MenuItem value={'outlined'} label={'outlined'}>
              outlined
            </MenuItem>
            <MenuItem value={'filled'} label={'filled'}>
              filled
            </MenuItem>
          </Select>
          <InputLabel>error</InputLabel>
          {/* <Checkbox label={"error"}/> */}
          <br />
          <hr />
          Components Test
          <div>
            <TestTextField />
          </div>
        </FormControl>
      </Paper>
    </div>
  )
}
