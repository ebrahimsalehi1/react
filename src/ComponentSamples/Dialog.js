
import React from 'react';
import TextField from "@material-ui/core/TextField";
import Grid from "@material-ui/core/Grid";
import Button from "@material-ui/core/Button";

import DialogNewComponent from "../Components/IrisaComponents/Dialog";

//import Dialog from '../components/common/IbxDialog';
import {Switch as SwitchUI} from "@material-ui/core";
import Typography from "@material-ui/core/Typography";
//import IbxDNDTree from "../components/common/IbxDNDTree";
import Card from "@material-ui/core/Card";
import IrisaButton from '../Components/IrisaComponents/Button/IrisaButton';

class MyDialog extends React.Component {
    constructor(props){
        super(props);
        this.state={
            open: false,
            checked: false,
        }
    }
    render() {
        return (
        <div>
            <button onClick={(e)=>{this.setState({open: true})}}>Show Dialog</button>
            <DialogNewComponent
                open={this.state.open}
                title={"Edit Customer"}
                maxWidth={"lg"}
                //fullScreen
                TransitionComponent
                eventClose={(e) => {
                    this.setState({open: false});
                }}
                actionBar={
                    <>
                        <Button color={"primary"} variant={"outlined"} style={{"width":"75px","margin":"2px"}}>Ok</Button>
                        <Button color={"primary"} variant={"outlined"} style={{"width":"75px","margin":"2px"}}>Cancel</Button>
                        <IrisaButton label={"Irisa Button"} variant={"outlined"} style={{"width":"75px","margin":"2px"}}/>
                    </>
                }
            >
                <Grid container spacing={16} >
                    <Grid item xs={6} md={6}>
                        <TextField  placeholder={"Email address"} variant={"outlined"}/>
                    </Grid>
                    <Grid item xs={6} md={6}>
                        <TextField  placeholder={"Full name"} variant={"outlined"}/>
                    </Grid>

                    <Grid item xs={6} md={6}>
                        <TextField  placeholder={"Phone Number"} variant={"outlined"}/>
                    </Grid>
                    <Grid item xs={6} md={6}>
                        <TextField  placeholder={"State/Region"} variant={"outlined"}/>
                    </Grid>

                    <Grid item xs={6} md={6}>
                        <TextField  placeholder={"Country"} variant={"outlined"}/>
                    </Grid>
                    <Grid item xs={6} md={6}>
                        <TextField  placeholder={"Address 1"} variant={"outlined"}/>
                    </Grid>

                    <Grid item xs={12} md={12}>
                        <TextField  placeholder={"Address 2"} variant={"outlined"}/>
                    </Grid>

                    <Grid item xs={6} md={6}>
                        <Typography gutterBottom variant={"h5"}>
                            Discounted Prices
                        </Typography>

                        <Typography gutterBottom component="h6" variant="h6">
                            This will give the user discountedprices for all products
                        </Typography>
                        <SwitchUI
                         checked={!this.state.checked}
                         value={"check box 1"}
                         color={"primary"}
                         onChange={(e)=>{
                             this.setState({checked:!this.state.checked});
                             console.log("checked 1",e);
                         }}
                        />
                    </Grid>
                    <Grid item xs={6} md={6}>

                        <Typography gutterBottom variant={"h5"}>
                            Email Verified
                        </Typography>

                        <Typography gutterBottom component="h6" variant="h6">
                            Disabling this will automatically send the user a
                            verification email
                        </Typography>
                        <SwitchUI
                         checked={this.state.checked}
                         value={"check box 2"}
                         color={"primary"}
                         onChange={(e)=>{
                             this.setState({checked:!this.state.checked});
                             console.log("checked 2",this.state.checked);
                         }}
                        />
                    </Grid>

                    <Grid item xs={6} md={6}>
                        <Typography variant={"h5"} gutterBottom>
                            The sample text in middle of form
                        </Typography>
                    </Grid>

                    <Grid item xs={6} md={6}>
                        <Typography variant={"h5"} gutterBottom>
                            This is a test
                        </Typography>
                    </Grid>

                    <Grid item xs={12} md={12}>
                        <Card style={{"min-height:":"100px","max-height:":"300px","overflow-y":"auto","border":"2px solid red"}}>
                        {/* <IbxDNDTree
                            data={
                              [{title:'A',children:[{title:'B',children:[{title:'B11'},{title:'B12',children:[{title:'B121'},{title:'B122'}]},{title:'B13'}]},{title:'C'}]}]
                            }/> */}
                        </Card>
                    </Grid>

                 </Grid>
            </DialogNewComponent>
        </div>
        );
    }
}

export default MyDialog;
