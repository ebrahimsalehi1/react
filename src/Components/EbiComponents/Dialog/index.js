import React,{memo} from 'react';
import Dialog from '@material-ui/core/Dialog';
import {withStyles} from "@material-ui/core";
import Typography from "@material-ui/core/Typography";
import PropTypes from "prop-types";
import Slide from "@material-ui/core/Slide";
import IconButton from "@material-ui/core/IconButton";
import {Close} from "@material-ui/icons";
import Card from "@material-ui/core/Card";
import Paper from "@material-ui/core/Paper";
import Divider from '@material-ui/core/Divider';
import Button from "@material-ui/core/Button";
import MuiDialogTitle from '@material-ui/core/DialogTitle';
import MuiDialogContent from '@material-ui/core/DialogContent';
import MuiDialogActions from '@material-ui/core/DialogActions';
import CloseIcon from '@material-ui/icons/Close';

const DialogTitle = withStyles(theme => ({
    root: {
        borderBottom: `1px solid ${theme.palette.divider}`,
        padding: '16px',
    },
    closeButton: {
        position: 'absolute',
        right: theme.spacing.unit,
        top: theme.spacing.unit,
        color: theme.palette.grey[500],
    }
}))(props => {
    const { children, classes, onClose } = props;
    return (
        <MuiDialogTitle disableTypography className={classes.root}>
            <Typography variant="h5" >{children}</Typography>
            {onClose ? (
                <IconButton aria-label="Close" className={classes.closeButton} onClick={onClose}>
                    <CloseIcon />
                </IconButton>
            ) : null}
        </MuiDialogTitle>
    );
});

const DialogActions = withStyles(theme => ({
    root: {
        borderTop: `1px solid ${theme.palette.divider}`,
        margin: "0px",
        padding:"8px",
    },
}))(MuiDialogActions);

const PaperDialogContent = withStyles(({
    root:{
        paddingTop:"16px",
        paddingLeft: "16px",
        paddingRight: "16px",
        paddingBottom: "16px",
        margin: 0,
        overflow: 'hidden',
        maxHeight: '600px',
        border:'0px',
        boxShadow: 0
    }
}))(Paper);

function Transition(props) {
    return <Slide direction="left" {...props} />;
}

class IbxDialog extends React.PureComponent {

    render() {
        const {classes, eventClose, maxWidth,
            openModal,title,children,fullScreen,
            TransitionComponent,actionBar,onKeyDown,useOwnDialog,...others} = this.props;

        return (
            <>
            {openModal &&
            <Dialog
                onClose={eventClose}
                onKeyDown={onKeyDown}
                data-testId
                aria-labelledby="customized-dialog-title"
                open={openModal}
                fullScreen={fullScreen}
                maxWidth={maxWidth}
                TransitionComponent={TransitionComponent ? Transition : null}
                {...others}
            >

                <DialogTitle id="customized-dialog-title" onClose={this.handleClose}>
                        {title}
                </DialogTitle>

                <muiDialogContent>

                    <PaperDialogContent>

                            {children}

                    </PaperDialogContent>

                </muiDialogContent>

                <DialogActions>
                    {actionBar}
                </DialogActions>

            </Dialog>
            }
            </>
        );
    }
}

IbxDialog.propTypes = {
    //classes: PropTypes.object.isRequired,
    eventClose: PropTypes.func.isRequired,
    openModal: PropTypes.bool.isRequired,
    title: PropTypes.string.isRequired,
    fullScreen:PropTypes.bool,
    TransitionComponent:PropTypes.bool,
    actionBar:PropTypes.node
};

export default IbxDialog;

