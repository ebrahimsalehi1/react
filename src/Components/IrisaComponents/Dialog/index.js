import React,{memo} from 'react';
import Dialog from '@material-ui/core/Dialog';
import {withStyles} from "@material-ui/core";
import Typography from "@material-ui/core/Typography";
//import MuiDialogTitle from '@material-ui/core/DialogTitle';
//import MuiDialogContent from '@material-ui/core/DialogContent';
//import {styles} from '../../assets/jss/style'
import PropTypes from "prop-types";
import Slide from "@material-ui/core/Slide";
//import DialogActions from "@material-ui/core/DialogActions";
import IconButton from "@material-ui/core/IconButton";
import {Close} from "@material-ui/icons";
import Card from "@material-ui/core/Card";
import Paper from "@material-ui/core/Paper";
import Divider from '@material-ui/core/Divider';
import Button from "@material-ui/core/Button";

import MuiDialogTitle from '@material-ui/core/DialogTitle';
import MuiDialogContent from '@material-ui/core/DialogContent';
import MuiDialogActions from '@material-ui/core/DialogActions';
import CloseIcon from '@material-ui/icons/Close';

/*
const DialogTitle = withStyles(theme => ({
    root: {
        borderBottom: `1px solid ${theme.palette.divider}`,
        textAlign:'center',
        margin: 0,
        padding: 0,//theme.spacing.unit*2,
        backgroundColor:'#009ce6',
        height:'20px'
    },
    closeButton: {
        position: 'absolute',
        right: '-2px',//theme.spacing.unit,
        top: '-2px',//theme.spacing.unit*2,
        color: theme.palette.common.white,
    },
}))(props => {
    const { children, classes, onClose } = props;
    return (
        <MuiDialogTitle disableTypography className={classes.root}>
            <Typography variant="h6">{children}</Typography>
            <IconButton aria-label="close" className={classes.closeButton} onClick={onClose}>
                <Close />
            </IconButton>
        </MuiDialogTitle>
    );
});
*/

const DialogTitle = withStyles(theme => ({
    root: {
        borderBottom: `1px solid ${theme.palette.divider}`,
    },
    closeButton: {
        position: 'absolute',
        right: theme.spacing.unit,
        top: theme.spacing.unit,
        color: theme.palette.grey[500],
    },
}))(props => {
    const { children, classes, onClose } = props;
    return (
        <MuiDialogTitle disableTypography className={classes.root}>
            <Typography variant="h6">{children}</Typography>
            {onClose ? (
                <IconButton aria-label="Close" className={classes.closeButton} onClick={onClose}>
                    <CloseIcon />
                </IconButton>
            ) : null}
        </MuiDialogTitle>
    );
});

// const DialogContent = withStyles(theme => ({
//     root: {
//         margin: 0,
//         padding: theme.spacing.unit * 2,
//     },
// }))(MuiDialogContent);

const DialogActions = withStyles(theme => ({
    root: {
        borderTop: `1px solid ${theme.palette.divider}`,
    },
}))(MuiDialogActions);

const PaperDialogContent = withStyles(({
    root:{
        paddingTop:"16px",
        paddingLeft: "16px",
        paddingRight: "16px",
        paddingBottom: "16px",
        margin: 0,
        overflowX: 'auto',
        overflowY: 'auto',
        maxHeight: '600px'
    }
}))(Paper);

/*
const DialogContent = withStyles(theme => ({
    root: {
        margin: 0,
        padding: 5,
        fontSize:11
    },
}))(MuiDialogContent);

const styles = {
    title:{
        display:"flex",
        padding: "16px",
        alignItems:"center",
    },
    footer:{
        display:"flex",
        padding: "8px",
        alignItems:"flex-end",
    }
};
*/
function Transition(props) {
    return <Slide direction="left" {...props} />;
}

class IbxDialog extends React.Component {

    render() {
        const {classes, eventClose, maxWidth,
            openModal,title,children,fullScreen,
            TransitionComponent,actionBar,onKeyDown,useOwnDialog,...others} = this.props;

        return (
                <Dialog
                    onClose={eventClose}
                    onKeyDown={onKeyDown}
                    aria-labelledby="customized-dialog-title"
                    open={openModal}
                    fullScreen={fullScreen}
                    maxWidth={maxWidth}
                    TransitionComponent={TransitionComponent?Transition:null}
                    {...others}
                >

                    <DialogTitle id="customized-dialog-title" onClose={this.handleClose}>
                        <Typography gutterBottom variant={"h5"}>
                            {title}
                        </Typography>
                    </DialogTitle>

                    <PaperDialogContent>

                        <muiDialogContent>

                            {children}

                        </muiDialogContent>

                        <DialogActions>
                            {actionBar}
                        </DialogActions>

                    </PaperDialogContent>


                </Dialog>

        );
    }
}

IbxDialog.propTypes = {
    classes: PropTypes.object.isRequired,
    eventClose: PropTypes.func.isRequired,
    openModal: PropTypes.bool.isRequired,
    title: PropTypes.string.isRequired,
    fullScreen:PropTypes.bool,
    TransitionComponent:PropTypes.bool,
    actionBar:PropTypes.node
};
export default IbxDialog;
//export default withStyles(styles)(IbxDialog);


