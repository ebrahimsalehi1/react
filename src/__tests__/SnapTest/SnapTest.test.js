import React from 'react';
import { mount ,shallow ,render } from '../../../enzymeConfig';
import Dialog from '../../ComponentSamples/Dialog';
//import {expect} from 'jest';
import { Calendar } from "react-modern-calendar-datepicker";
import TextField from '@material-ui/core/TextField';
import DialogMaterial from '@material-ui/core/Dialog';
import IrisaButton from '../../Components/EbiComponents/Button/index';
import renderer from 'react-test-renderer';

describe('main test',()=>{
    // const MyComp = ()=>(
    //     <div>
    //         <label>This is a test</label>            
    //     </div>
    // )
    // it('test1',()=>{


    //     const TestedComp = renderer.create(<MyComp/>).toJSON();
    //     expect(TestedComp).toMatchSnapshot();

    // });

    it('4. Test events - sample component',()=>{
        const MyComp = () =>(
            <div>
                <h1>my title 1</h1>
                <input type={'TEXT'} />
                <button id={'button1'} onClick={e=>{console.log('click')}}>Click me</button>
            </div>
        );

        const onClick = jest.fn();
        const comp1 = render(<button id={'button1'} onClick={onClick}>Click me</button>);
        //comp1.find('#button1').simulate('click');
        comp1.simulate('click');

        expect(onClick).toHaveBeenCalled();
    });



})