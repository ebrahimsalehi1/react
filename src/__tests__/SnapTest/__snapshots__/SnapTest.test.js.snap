// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main test test1 1`] = `
<div>
  <label>
    This is a test
  </label>
</div>
`;
